static/src/js/util/request/openSearchGranuleRequest.js

```diff
@@ -175,20 +175,26 @@ export default class OpenSearchGranuleRequest extends Request {
         let browseUrl
 
         granuleLinks.filter(Boolean).forEach((link) => {
-          if (link.rel === 'icon') {
-            updatedGranule.thumbnail = link.href
-
-            // We found a suitable image, flip the flag to true
-            updatedGranule.browse_flag = true
-          }
-
-          // CWIC does not have a standard link relation for large-sized browse,
-          // so we look for URLs with extensions of jpg, jpeg, gif, and png, and
-          // return the first one which is not marked as the icon. If there's
-          // no such relation, we return nothing, which results in there being
-          // no link to the full size image.
-          if (!browseUrl && link.rel !== 'icon' && link.href.match(/\.(?:gif|jpg|jpeg|png)(?:\?|#|$)/)) {
-            browseUrl = link.href
+          if (isString(link)) {
+            if (link.match(/\.(?:gif|jpg|jpeg|png)(?:\?|#|$)/)) {
```

Comment:
Lets make a quick util for this instead of duplicating it -- can even live in this class as a method.